src/asm/lexer.c

```diff
@@ -921,6 +921,31 @@ void lexer_DumpStringExpansions(void)
 	free(stack);
 }
 
+/* Discards an inline comment */
+static void discardInlineComment(void)
+{
+	dbgPrint("Discarding inline comment\n");
+	int c;
+	shiftChars(2);
+	for (;;) {
+		c = peek(0);
```

Comment:
Arguably, this can become a `switch (nextChar())`. Since that shifts the character being matched, the following `peek`s get an argument of 0, which is better. (You'd move the `shiftChars(1)` at the end of the loop to after it, too.)